cmd/swarm/swarm-smoke-pss/pss.go

```diff
@@ -0,0 +1,340 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"errors"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/metrics"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/go-ethereum/swarm/pss"
+	"github.com/ethereum/go-ethereum/swarm/testutil"
+
+	cli "gopkg.in/urfave/cli.v1"
+)
+
+type pssMode int
+
+const (
+	pssModeRaw = iota
+	pssModeAsym
+	pssModeSym
+)
+
+func (m pssMode) String() string {
+	return [...]string{"raw", "asym", "sym"}[m]
+}
+
+type pssJob struct {
+	sender   *pssNode
+	receiver *pssNode
+	mode     pssMode
+	msg      []byte
+}
+
+type pssNode struct {
+	hostIdx        int
+	addr           []byte
+	pubkey         string
+	client         *rpc.Client
+	deregisterFunc func()
+}
+
+type pssSession struct {
+	topic pss.Topic
+	msgC  chan pss.APIMsg
+	nodes []*pssNode
+	jobs  map[string]*pssJob
+}
+
+func pssAsymCheck(ctx *cli.Context, tuid string) error {
```

Comment:
nitpick, but i introduced `tuid` as a `test uuid` in order to get all logs from a specific test. this is no longer needed, as i've noticed we can filter by the job id on elastic search, so `tuid` is completely redundant now. also it is not passed to the `runCheck` function.

feel free to remove or leave as is for now.

Reply:
I've removed it.